scripts/main/mapview.js

```diff
@@ -2,48 +2,98 @@
  * @description This module takes care of the map view of a full album and its sub-albums.
  */
 
+/**
+ * @typedef MapProvider
+ * @property {string} layer - URL pattern for map tile
+ * @property {string} attribution - HTML with attribution
+ */
+
 const map_provider_layer_attribution = {
+	/**
+	 * @type {MapProvider}
+	 */
 	Wikimedia: {
 		layer: "https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}{r}.png",
 		attribution: '<a href="https://wikimediafoundation.org/wiki/Maps_Terms_of_Use">Wikimedia</a>',
 	},
+	/**
+	 * @type {MapProvider}
+	 */
 	"OpenStreetMap.org": {
 		layer: "https://{s}.tile.osm.org/{z}/{x}/{y}.png",
 		attribution: '&copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors',
 	},
+	/**
+	 * @type {MapProvider}
+	 */
 	"OpenStreetMap.de": {
 		layer: "https://{s}.tile.openstreetmap.de/{z}/{x}/{y}.png ",
 		attribution: '&copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors',
 	},
+	/**
+	 * @type {MapProvider}
+	 */
 	"OpenStreetMap.fr": {
 		layer: "https://{s}.tile.openstreetmap.fr/osmfr/{z}/{x}/{y}.png ",
 		attribution: '&copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors',
 	},
+	/**
+	 * @type {MapProvider}
+	 */
 	RRZE: {
 		layer: "https://{s}.osm.rrze.fau.de/osmhd/{z}/{x}/{y}.png",
 		attribution: '&copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors',
 	},
 };
 
-let mapview = {
+const mapview = {
+	/** @type {?L.Map} */
 	map: null,
 	photoLayer: null,
+	/** @type {?number} */
 	min_lat: null,
+	/** @type {?number} */
 	min_lng: null,
+	/** @type {?number} */
 	max_lat: null,
+	/** @type {?number} */
 	max_lng: null,
+	/** @type {?string} */
 	albumID: null,
+	/** @type {?string} */
 	map_provider: null,
 };
 
+/**
+ * @typedef MapPhotoEntry
+ *
+ * @property {number} [lat] - latitude
+ * @property {number} [lng] - longitude
+ * @property {string} [thumbnail] - URL to the thumbnail
+ * @property {string} [thumbnail2x] - URL to the high-res thumbnail
+ * @property {string} url - URL to the small size-variant; quite a misnomer
+ * @property {string} url2x - URL to the small, high-res size-variant; quite a misnomer
+ * @property {string} name - the title of the photo
+ * @property {string} taken_at - the takedate of the photo, formatted as a locale string
+ * @property {string} albumID - the album ID
+ * @property {string} photoID - the photo ID
+ */
+
+/**
+ * @returns {boolean}
+ */
 mapview.isInitialized = function () {
-	if (mapview.map === null || mapview.photoLayer === null) {
-		return false;
-	}
-	return true;
+	return !(mapview.map === null || mapview.photoLayer === null);
 };
 
+/**
+ * @param {?string} _albumID
+ * @param {string} _albumTitle
+ *
+ * @returns {void}
+ */
 mapview.title = function (_albumID, _albumTitle) {
+	// TODO: Where are these single-letter IDs for builtin-smart albums defined?
```

Comment:
Legacy from v3, probably forgot to rename them:
f = favorite
u = unsorted
r = recent
p = public
if I remember correctly.

Reply:
Well the clue is just below this line ;) (I only opened it because I was pretty sure unsorted was 0)

Reply:
Actually, it's not just from v3, but from early v4 releases. I believe it was merged in 4.0.7: https://github.com/LycheeOrg/Lychee/pull/625

Reply:
> Well the clue is just below this line ;) (I only opened it because I was pretty sure unsorted was 0)

What line below do you mean? The line below is `switch(_albumID)`. That doesn't explain where the single letter IDs come from.

> Legacy from v3, probably forgot to rename them:
> f = favorite
> u = unsorted
> r = recent
> p = public

That is something which I also have inferred myself. That is rather obvious if you look at what the different `case`s do. My question was not, what the single letters mean. My question was where are the _defined_ in the sense of computer programming, i.e. like in function declaration vs. function _definition_. My main point is, IMHO the `albumID` is never equal to one of these single letters. So why does this code work? IMHO, none of these cases should ever be catched by the program flow. But we also did not get any bug report. This might indicate that this piece of code is never called which means that this is probably dead code.

Reply:
I don't think it works. It needs to be fixed. Probably few people use the "map view" feature, especially in combination with smart albums.

Reply:
Fixed it.